components/FourPicsTable.js

```diff
@@ -1,47 +1,70 @@
 import { StyledImage } from "./StyledImage";
 import { StyledA } from "./StyledLink";
 import styled from "styled-components";
+import { useRouter } from "next/router";
+import useSWR from "swr";
+
+const fetcher = (url) => fetch(url).then((res) => res.json());
 
 export default function FourPicsTable() {
+  const router = useRouter();
+  const { searchID } = router.query;
+
+  const {
+    data: shirts,
+    isLoading,
+    error,
+  } = useSWR(`/api/${searchID}`, fetcher);
+
+  if (isLoading) {
+    return <div>loading...</div>;
+  }
+  if (error) {
+    return <div>error...</div>;
+  }
   return (
-    <FourPicsContainer>
-      <StyledA href="/PreviewPage/">
-        <StyledImage
-          oneOfFourPictures
-          alt="image number 1"
-          src="https://i.imgur.com/8nUXl6s.jpeg/"
-          width="400"
-          height="400"
-        />
-      </StyledA>
-      <StyledA href="/PreviewPage/">
-        <StyledImage
-          oneOfFourPictures
-          alt="image number 2"
-          src="https://i.imgur.com/SglfvOg.jpeg/"
-          width="400"
-          height="400"
-        />
-      </StyledA>
-      <StyledA href="/PreviewPage/">
-        <StyledImage
-          oneOfFourPictures
-          alt="image number 3"
-          src="https://i.imgur.com/rTEFFzZ.webp?maxwidth=1520&fidelity=grand/"
-          width="400"
-          height="400"
-        />
-      </StyledA>
-      <StyledA href="/PreviewPage/">
-        <StyledImage
-          oneOfFourPictures
-          alt="image number 4"
-          src="https://i.imgur.com/SME5yCA.jpeg/"
-          width="400"
-          height="400"
-        />
-      </StyledA>
-    </FourPicsContainer>
+    <>
+      {shirts.map((shirt) => (
+        <FourPicsContainer key={shirt._id}>
+          <StyledA href={`/PreviewPage/${shirt.pic1.picSRCSlug}`}>
+            <StyledImage
+              alt="image number 1"
+              src={shirt.pic1.picSRC}
+              width="400"
+              height="400"
+              $fourpictures
```

Comment:
We should aim for consistency of prop order. In 3 of the images `fourpictures` is the first prop you pass to the `StyledImage` but in this case it is the last prop. (This is a minor point and won't affect the outcome, just the readability of the code)